test/e2e/search_test.go

```diff
@@ -710,5 +710,72 @@ var _ = ginkgo.Describe("[karmada-search] karmada search testing", ginkgo.Ordere
 				})
 			})
 		})
+		ginkgo.Describe("update cluster testing", ginkgo.Ordered, func() {
+			ginkgo.Context("cluster labeled after registry created", ginkgo.Ordered, func() {
+				const labelKey = "cluster.karmada.io/managed"
+				var clusterLabels = map[string]string{labelKey: "true"}
+				var registryName string
+				var registry *searchv1alpha1.ResourceRegistry
+				ginkgo.BeforeAll(func() {
+					registryName = "rr-" + rand.String(RandomStrLength)
+					registry = &searchv1alpha1.ResourceRegistry{
+						ObjectMeta: metav1.ObjectMeta{
+							Name: registryName,
+						},
+						Spec: searchv1alpha1.ResourceRegistrySpec{
+							TargetCluster: policyv1alpha1.ClusterAffinity{
+								LabelSelector: &metav1.LabelSelector{
+									MatchExpressions: []metav1.LabelSelectorRequirement{{
+										Key:      labelKey,
+										Operator: metav1.LabelSelectorOpExists,
+									}},
+								},
+							},
+							ResourceSelectors: []searchv1alpha1.ResourceSelector{
+								{
+									APIVersion: "v1",
+									Kind:       "Node",
+								},
+							},
+						},
+					}
+					framework.CreateResourceRegistry(karmadaClient, registry)
+				})
+				ginkgo.AfterAll(func() {
+					framework.RemoveResourceRegistry(karmadaClient, registryName)
+					framework.DeleteClusterLabels(karmadaClient, member1, clusterLabels)
+				})
+				ginkgo.It("cluster update trigger reconcile", func() {
+					// The cluster didn't match the label selector of registry
+					var proxyList *corev1.NodeList
+					gomega.Eventually(func(g gomega.Gomega) {
+						var err error
+						proxyList, err = proxyClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
```

Comment:
The current modifications made in PR should not involve any changes in the search/proxy functionality. This test example may have been placed in the wrong location and can be moved to this side of line 344.